### DISCUSSION TITLE
Unofficial NZP Launcher for Linux

### DISCUSSION THREAD

**Discussion by Paindeluxe** [nzp-team/nzportable#1064] [Other Content Releases]
Hello everyone, I just made a NZP Launcher which automatically Updates NZP with Mod Support for Linux.  
I also plan on making it compatible with Windows and maybe even MacOS.
I hope y'all like it, and feel free to give me Feedback if you got any!

https://github.com/Paindeluxe/Unofficial-NZP-Launcher

**Comment by Paindeluxe** [comment]
I am aware that a NZP Launcher for Linux already exists, but it's only for Arch Linux.
This one is hopefully universal as it's a Bash Script.

**Comment by StimPak98** [comment]
Thank you so much! I really enjoy your launcher as it makes modding and launch NZP so much easier! Thank you!

**Comment by BrownNPC** [comment]
You could easily translate this code to Lua to make it cross platform. Or python.
I do recommend python because it's pre installed on all Unix operating systems and for windows you could compile it to an exe

**Reply by Paindeluxe** [reply]
While I agree that both of your proposed alternative Programming Languages are good alternatives to Bash. Python and Lua are not always preinstalled on Linux Devices and in my Opinion take up a generous Amount of Space to install compared to Bash or zsh. (If I am not mistaken.)

I plan on learning C or C++ after reaching Version 2.0.0 of this Launcher and rewriting it in that.

And to explain why I first programmed this in Bash... well it seemed easier than learning C?

**Reply by BrownNPC** [reply]
Actually python is pre installed in most distributions except for maybe arch. Like it's in Ubuntu, Mint, Fedora, and Debian. 
Also on MacOS.

If you're still thinking of using a compiled language then Maybe you can try a simple compiled language Like Go. Which has cross compilation out of the box.

Although the python solution is the simplest :)

**Reply by katniny** [reply]
why's it matter so much what he uses? :p 
just fork it and do it yourself if you're that concerned about it

**Reply by BrownNPC** [reply]
> why's it matter so much what he uses? :p just fork it and do it yourself if you're that concerned about it

Everything I said has been said from my experience and is an opinion. 

OP said he wishes to make the script cross platform, so I mentioned that it would be really easy to port it to python and get what he achieves. 

I have the best of intent, and I'm not trying to advocate for a programming language just for the sake of it. 

At the end of the day it's OP's choice. But I assume this is an open discussion board free to people's opinions if it does not derail from the original topic.

As far as the DIY approach goes, I could definitely do that. I'm sure I could even put this script into chat gpt and get it translated into a different language. 
But again, I'm not trying to advocate for a language for the sake of it.
Why would I take someone's fun away of building their project by forking and making a "better" version. I'm not trying to arouse my ego or anything lol.

I hope that is a detailed enough answer, have a good day 😊

**Reply by Paindeluxe** [reply]
> just fork it and do it yourself if you're that concerned about it

> Why would I take someone's fun away of building their project by forking and making a "better" version. I'm not trying to arouse my ego or anything lol.

I wouldn't mind it at all, I'm just happy to see People using what I make.

 > Maybe you can try a simple compiled language Like Go. Which has cross compilation out of the box.

I didn't know Go compiles to Code, I'll make sure to check it out when I can!
